packages/opentelemetry-resources/src/platform/node/detectors/AwsEc2Detector.ts

```diff
@@ -44,41 +48,86 @@ class AwsEc2Detector implements Detector {
    */
   async detect(config: ResourceDetectionConfigWithLogger): Promise<Resource> {
     try {
+      const token = await this._fetchToken();
       const {
         accountId,
         instanceId,
         instanceType,
         region,
         availabilityZone,
-      } = await this._awsMetadataAccessor();
+      } = await this._fetchIdentity(token);
+      const hostname = await this._fetchHost(token);
+
       return new Resource({
         [CLOUD_RESOURCE.PROVIDER]: 'aws',
         [CLOUD_RESOURCE.ACCOUNT_ID]: accountId,
         [CLOUD_RESOURCE.REGION]: region,
         [CLOUD_RESOURCE.ZONE]: availabilityZone,
         [HOST_RESOURCE.ID]: instanceId,
         [HOST_RESOURCE.TYPE]: instanceType,
+        [HOST_RESOURCE.NAME]: hostname,
+        [HOST_RESOURCE.HOSTNAME]: hostname,
       });
     } catch (e) {
       config.logger.debug(`AwsEc2Detector failed: ${e.message}`);
       return Resource.empty();
     }
   }
 
+  private async _fetchToken(): Promise<string> {
+    const options = {
+      host: this.AWS_IDMS_ENDPOINT,
+      path: this.AWS_INSTANCE_TOKEN_DOCUMENT_PATH,
+      method: 'PUT',
+      timeout: 1000,
+      headers: {
+        'X-aws-ec2-metadata-token-ttl-seconds': '60',
```

Comment:
can you please define some const for that header name ?

Reply:
One question here is how to assign this value, for instance I did
```
readonly AWS_METADATA_TTL_HEADER = 'X-aws-ec2-metadata-token-ttl-seconds';
headers: {
   this.AWS_METADATA_TTL_HEADER: '60',
},
```
I also tried to build a global ```AWS_METADATA_TTL_HEADER```, it also didnt work. It seems that Javascript always take these variable names as key value rather than extract the value of variables.
Any suggestions?

Reply:
I just found a way like we could
```
const entries = new Map([
  [this.AWS_METADATA_TTL_HEADER, '60'],
]);
header: Object.fromEntries(entries);
```
But I dont think it looks better than the original one

Reply:
Also, if we use ```Object.fromEntries()``` method, we need to introduce ```lib: ["es2019"]``` in our compiler cofiguration

Reply:
don't try to overengineering :) 

```
const SOME_APPROPRIATE_VALUE_NAME =  'X-aws-ec2-metadata-token-ttl-seconds';

/// .....
headers: {
  [SOME_APPROPRIATE_VALUE_NAME]: 60,
}
```